### PR TITLE
Store FastMapHistogram counts as Longs

### PR DESCRIPTION
**Pull request by jpolchlo**
FastMapHistogram was storing items and counts in a single array, with element 2\*i holding the item and 2\*i+1 storing the count.  Counts could roll over for values above Int.MaxValue.  This change splits the items and counts into two arrays, where the latter now stores Long values.

### CODE REVIEW

File: raster/src/main/scala/geotrellis/raster/histogram/FastMapHistogram.scala [hunk 1]
```diff
@@ -58,9 +60,9 @@ object FastMapHistogram {
   * The [[FastMapHistogram]] class.  Quickly compute histograms over
   * integer data.
   */
-class FastMapHistogram(_size: Int, _buckets: Array[Int], _used: Int, _total: Int)
+class FastMapHistogram(var size: Int, var buckets: Array[Int], var counts: Array[Long], var used: Int, var total: Long)
```
Comment (echeipesh) [inline raster/src/main/scala/geotrellis/raster/histogram/FastMapHistogram.scala]: This exposes `size`, `buckets`, `counts` as mutable fields, they should be private.

```scala
scala> class Box(var x: Int)
defined class Box

scala> val b = new Box(3)
b: Box = Box@4a1f53f1

scala> b.x
res0: Int = 3

scala> b.x = 5
b.x: Int = 5

scala> b.x
res1: Int = 5
```

Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/histogram/FastMapHistogram.scala]: That's my bad, I gave @jpolchlo false info.

Comment (echeipesh) [inline raster/src/main/scala/geotrellis/raster/histogram/FastMapHistogram.scala]: Other than that this looks good to me. Precarious +1 before Travis.

Comment (echeipesh) [inline raster/src/main/scala/geotrellis/raster/histogram/FastMapHistogram.scala]: Also your commits are not registered with the github account: https://github.com/geotrellis/geotrellis/pull/1455/commits/8d517fea7dc2d6869815bd4262ec4706500d55a8

Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/histogram/FastMapHistogram.scala]: That's important to fix, you have to set up your global email settings on your machine to your github account email so they link up.

**Comment by echeipesh** [comment]
Superseded by https://github.com/geotrellis/geotrellis/pull/1457

**Pull request closed without merging**